src/community/script/core/src/main/java/org/geoserver/script/app/AppListResource.java

```diff
@@ -35,23 +33,13 @@ public AppListResource(ScriptManager scriptMgr, Request request, Response respon
     }
 
     public void handleGet() {
-        File appRoot;
-        try {
-            appRoot = scriptMgr.getAppRoot();
-        } catch (IOException e) {
-            throw new RestletException("Error looking up apps", Status.SERVER_ERROR_INTERNAL, e);
-        }
+        org.geoserver.platform.resource.Resource appRoot = scriptMgr.app();
 
-        File[] apps = appRoot.listFiles(new FileFilter() {
-            @Override
-            public boolean accept(File pathname) {
-                return pathname.isDirectory();
-            }
-        });
+        List<org.geoserver.platform.resource.Resource> apps = Resources.list(appRoot, Resources.DirectoryFilter.INSTANCE);
```

Comment:
This is great. The new API just cleans everything up. Great work @NielsCharlier.